wordpress-mobile/WordPress-Android
# Add View All screen for Annual stats

Fixes #9647 
This PR adds a "View more" button to the "This year" stats card. Once the button is clicked, the "View all" page opens with a year picker so the user can look at the previous years in stats.

This View all screen is a special use case since it's basically an "Insights" use case that also needs to show the date picker. It only has one granularity - YEARS. I've added a separate case to the `StatsViewAllViewModelFactory`. Let me know what you think about it @0nko 

The "View all" page has a different designs (everything on separate lines).

To test:
* Go to Stats/Insights for a site that is older than 1 year
* Scroll to "This year" card
* Notice there is a "View more" button on the bottom of the card
* Click on the "View more" button
* The "View all" page opens
* There is a year picker on top of the screen
* Changing years works

![Screenshot_1555501358](https://user-images.githubusercontent.com/1079756/56285111-b546c680-6116-11e9-8b32-158827ca025f.png)
